test/ServerRequestFactoryTest.php

```diff
@@ -20,6 +20,18 @@
 
 class ServerRequestFactoryTest extends TestCase
 {
+    protected $globalServer;
+
+    protected function setUp()
+    {
+        $this->globalServer = $_SERVER;
+    }
+
+    protected function tearDown()
+    {
+        $_SERVER = $this->globalServer;
+    }
+
```

Comment:
Why is this functionality needed? The `@backupGlobals` and `@preserveGlobalState` annotations are `enabled` by default, which should make this unnecessary. What did you observe that led to their addition?